exporter/datadogexporter/factory.go

```diff
@@ -190,10 +190,9 @@ func NewFactory() exporter.Factory {
 }
 
 func defaultClientConfig() confighttp.ClientConfig {
-	// do not use NewDefaultClientConfig for backwards-compatibility
```

Comment:
Does this still apply? What are the defaults in confighttp.NewDefaultClientConfig()?

Reply:
The defaults are the defaultTransport defaults ([see here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L116-L133)). Even if set to nil, when [ToClient is called](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L136) it uses the defaultTransport defaults anyways. The only difference is the Headers field used to be nil, but now it's an empty map. This is why I changed the validation in `config.go` to empty from nil. 

That said, I'm not sure what specifics we were trying to maintain for backwards-compatibility, but the change here doesn't change the behaviour.